src/ducks/apps/index.js

```diff
@@ -411,34 +411,53 @@ export function fetchInstalledApps(lang, fetchingRegistry) {
   return async dispatch => {
     try {
       // Start the HTTP requests as soon as possible
-      let fetchingKonnectors = cozy.client.fetchJSON('GET', '/konnectors/')
-      let fetchingWebApps = cozy.client.fetchJSON('GET', '/apps/')
+      let fetchingKonnectors = null
+      let fetchingWebApps = null
+      if (
+        !CONSTANTS.filterAppType ||
+        CONSTANTS.filterAppType === APP_TYPE.KONNECTOR
+      ) {
+        fetchingKonnectors = cozy.client.fetchJSON('GET', '/konnectors/')
+      }
+      if (
+        !CONSTANTS.filterAppType ||
+        CONSTANTS.filterAppType === APP_TYPE.WEBAPP
+      ) {
+        fetchingWebApps = cozy.client.fetchJSON('GET', '/apps/')
+      }
       await fetchingRegistry
       dispatch({ type: FETCH_APPS })
-      let installedWebApps = await fetchingWebApps
-      installedWebApps = installedWebApps.map(w => {
-        // FIXME type konnector is missing from stack
-        w.attributes.type = APP_TYPE.WEBAPP
-        return w
-      })
-      // TODO throw error if collect is not installed
-      const collectApp = installedWebApps.find(
-        a => a.attributes.slug === 'collect'
-      )
-      const collectLink = collectApp && collectApp.links.related
-      installedWebApps = installedWebApps.filter(
-        app => !config.notDisplayedApps.includes(app.attributes.slug)
-      )
-      let installedKonnectors = await fetchingKonnectors
-      installedKonnectors = installedKonnectors.map(k => {
-        // FIXME type konnector is missing from stack
-        k.attributes.type = APP_TYPE.KONNECTOR
-        return k
-      })
-      installedKonnectors = installedKonnectors.filter(
-        app => !config.notDisplayedApps.includes(app.attributes.slug)
-      )
-      const installedApps = installedWebApps.concat(installedKonnectors)
+      let installedApps = []
+      let collectLink
+      if (fetchingWebApps) {
+        let installedWebApps = await fetchingWebApps
+        installedWebApps = installedWebApps.map(w => {
+          // FIXME type konnector is missing from stack
+          w.attributes.type = APP_TYPE.WEBAPP
+          return w
+        })
+        // TODO throw error if collect is not installed
+        const collectApp = installedWebApps.find(
```

Comment:
Maybe it's time to rename `collectApp` to `homeApp`

Reply:
Even better, we don't need this collectApp/Link anymore since we use the intents now, thanks for this point 👍